### PR TITLE
map new packets and general networking

### PR DESCRIPTION
**Pull request by eristhea**
5000 new class attack you die forever.
~~still WIP, I want to 100% the `network` package~~

### CODE REVIEW

**Review by EnnuiL — APPROVED** [review]
Looks good so far!

File: mappings/net/minecraft/network/packet/s2c/payload/GameEventListenersDebugPayload.mapping [hunk 1]
```diff
@@ -0,0 +1,2 @@
+CLASS net/minecraft/unmapped/C_jcphmavb net/minecraft/network/packet/s2c/payload/GameEventListenersDebugPayload
+	FIELD f_pwweslns OD Lnet/minecraft/unmapped/C_ncpywfca;
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/network/packet/s2c/payload/GameEventListenersDebugPayload.mapping]:
```suggestion
	FIELD f_pwweslns ID Lnet/minecraft/unmapped/C_ncpywfca;
```

File: mappings/net/minecraft/network/packet/s2c/payload/StructuresDebugPayload.mapping [hunk 2]
```diff
@@ -0,0 +1,10 @@
+CLASS net/minecraft/unmapped/C_djjgdtqh net/minecraft/network/packet/s2c/payload/StructuresDebugPayload
+	FIELD f_kdybrwcv ID Lnet/minecraft/unmapped/C_ncpywfca;
+	METHOD m_akcfvyrp writeBlockBox (Lnet/minecraft/unmapped/C_idfydwco;Lnet/minecraft/unmapped/C_hmqypqzz;)V
+		ARG 1 box
+	METHOD m_ohxxgyup readBlockBox (Lnet/minecraft/unmapped/C_idfydwco;)Lnet/minecraft/unmapped/C_hmqypqzz;
+	METHOD m_yfypquja (Lnet/minecraft/unmapped/C_idfydwco;Lnet/minecraft/unmapped/C_idfydwco;Lnet/minecraft/unmapped/C_djjgdtqh$C_obwdgvyc;)V
+		ARG 1 byteBuf
+		ARG 2 piece
+	CLASS C_obwdgvyc PieceQWE
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/network/packet/s2c/payload/StructuresDebugPayload.mapping]:
```suggestion
	CLASS C_obwdgvyc PieceQwe
```

**Review by supersaiyansubtlety — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/network/packet/s2c/payload/StructuresDebugPayload.mapping (hunk 2 above)
Comment (eristhea) [reply mappings/net/minecraft/network/packet/s2c/payload/StructuresDebugPayload.mapping]: I need to release my fix hang on five seconds. once again working around the enigma name conflict bug with that one

**Review by EnnuiL — APPROVED** [review]

File: mappings/net/minecraft/network/ConnectionIntent.mapping [hunk 3]
```diff
@@ -0,0 +1,7 @@
+CLASS net/minecraft/unmapped/C_uxoyljpb net/minecraft/network/ConnectionIntent
+	FIELD f_ipgpjumr LOGIN_ID I
+	FIELD f_jclclbta STATUS_ID I
+	METHOD m_urowrtvw byId (I)Lnet/minecraft/unmapped/C_uxoyljpb;
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/network/ConnectionIntent.mapping]:
```suggestion
	METHOD m_urowrtvw get (I)Lnet/minecraft/unmapped/C_uxoyljpb;
```
I think `byX` is typically for getting a map-like thing, not getting a value from a map-like thing.

File: mappings/net/minecraft/network/PacketByteBuf.mapping [hunk 4]
```diff
@@ -272,6 +272,12 @@ CLASS net/minecraft/unmapped/C_idfydwco net/minecraft/network/PacketByteBuf
 		COMMENT @see #readProperty()
 		ARG 1 property
 	METHOD m_abmmdpgc writePropertyMap (Lcom/mojang/authlib/properties/PropertyMap;)V
+		COMMENT Writes a map of properties to this buf.
+		COMMENT A property is represented by two strings, indicating the key and value, followed by a boolean indicating if a
+		COMMENT a signature is present, followed by a string indicating the signature only if it's present.
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/network/PacketByteBuf.mapping]:
```suggestion
		COMMENT A property is represented by two strings indicating the key and the value, followed by a boolean indicating if a
		COMMENT signature is present, and if present, followed by a string indicating the signature.
```

File: mappings/net/minecraft/network/PacketByteBuf.mapping [hunk 5]
```diff
@@ -601,6 +609,12 @@ CLASS net/minecraft/unmapped/C_idfydwco net/minecraft/network/PacketByteBuf
 		COMMENT @return this buf, for chaining
 		ARG 1 array
 			COMMENT the array to write
+	METHOD m_ngsogwlc decodeById (Ljava/util/function/IntFunction;)Ljava/lang/Object;
+		COMMENT Reads a {@linkplain #readVarInt var int} from this buf,
+		COMMENT and then runs {@code idToValue} to convert it to a value.
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/network/PacketByteBuf.mapping:613]: seems strange to talk about implementation here, maybe tag with `@implNote` if the implementation really needs to be discussed

File: mappings/net/minecraft/network/PacketByteBuf.mapping [hunk 6]
```diff
@@ -688,6 +702,12 @@ CLASS net/minecraft/unmapped/C_idfydwco net/minecraft/network/PacketByteBuf
 		ARG 1 bitSet
 			COMMENT the bit set to write
 	METHOD m_rfxcdbdu readPropertyMap ()Lcom/mojang/authlib/properties/PropertyMap;
+		COMMENT Reads a map of properties from this buf.
+		COMMENT A property is represented by two strings, indicating the key and value, followed by a boolean indicating if a
+		COMMENT a signature is present, followed by a string indicating the signature only if it's present.
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/network/PacketByteBuf.mapping]:
```suggestion
		COMMENT A property is represented by two strings indicating the key and the value, followed by a boolean indicating if a
		COMMENT signature is present, and if present, followed by a string indicating the signature.
```

File: mappings/net/minecraft/network/SplitterHandler.mapping [hunk 7]
```diff
@@ -1,5 +1,12 @@
 CLASS net/minecraft/unmapped/C_hoasxhkp net/minecraft/network/SplitterHandler
-	METHOD decode decode (Lio/netty/channel/ChannelHandlerContext;Lio/netty/buffer/ByteBuf;Ljava/util/List;)V
+	FIELD f_bmaoystr MAX_LENGTH_BYTES I
+	FIELD f_rvhsntis helperBuf Lio/netty/buffer/ByteBuf;
+	METHOD decode (Lio/netty/channel/ChannelHandlerContext;Lio/netty/buffer/ByteBuf;Ljava/util/List;)V
 		ARG 1 context
 		ARG 2 buffer
 		ARG 3 output
+	METHOD handlerRemoved0 (Lio/netty/channel/ChannelHandlerContext;)V
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/network/SplitterHandler.mapping:8]:
```suggestion
	METHOD handlerRemoved (Lio/netty/channel/ChannelHandlerContext;)V
```
?

File: mappings/net/minecraft/util/network/LegacyQueryUtil.mapping [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+	FIELD f_kqepobnp CUSTOM_PAYLOAD_PACKET_ID I
+	FIELD f_oifrnwvr CUSTOM_PAYLOAD_PACKET_PING_CHANNEL Ljava/lang/String;
+	FIELD f_pdmxsxoh FAKE_PROTOCOL_VERSION I
+	FIELD f_ppbxvghx DISCONNECT_PACKET_ID I
+	FIELD f_sccvugza QUERY_PACKET_ID I
+	METHOD m_fysytacf readString (Lio/netty/buffer/ByteBuf;Ljava/lang/String;)V
+		ARG 0 buf
+		ARG 1 string
+	METHOD m_hczwptxm writeString (Lio/netty/buffer/ByteBuf;)Ljava/lang/String;
+		ARG 0 buf
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/util/network/LegacyQueryUtil.mapping]:
```suggestion
	METHOD m_fysytacf writeString (Lio/netty/buffer/ByteBuf;Ljava/lang/String;)V
		ARG 0 buf
		ARG 1 string
	METHOD m_hczwptxm readString (Lio/netty/buffer/ByteBuf;)Ljava/lang/String;
		ARG 0 buf
```

**Review by supersaiyansubtlety — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/network/SplitterHandler.mapping (hunk 7 above)
Comment (eristhea) [reply mappings/net/minecraft/network/SplitterHandler.mapping:8]: automapped!

File: mappings/net/minecraft/network/PacketByteBuf.mapping (hunk 5 above)
Comment (eristhea) [reply mappings/net/minecraft/network/PacketByteBuf.mapping:613]: I think it needs the information, since you're passing in a function

**Review by supersaiyansubtlety — APPROVED** [review]

File: mappings/net/minecraft/network/SplitterHandler.mapping (hunk 7 above)
Comment (supersaiyansubtlety) [reply mappings/net/minecraft/network/SplitterHandler.mapping:8]: Why was this reverted?

Comment (eristhea) [reply mappings/net/minecraft/network/SplitterHandler.mapping:8]: this is not a name for the method, this is its obf name

Comment (supersaiyansubtlety) [reply mappings/net/minecraft/network/SplitterHandler.mapping:8]: oooh, oops

File: mappings/net/minecraft/network/packet/s2c/payload/WorldgenAttemptDebugPayload.mapping [hunk 9]
```diff
@@ -0,0 +1,2 @@
+CLASS net/minecraft/unmapped/C_djufnofa net/minecraft/network/packet/s2c/payload/WorldgenAttemptDebugPayload
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/network/packet/s2c/payload/WorldgenAttemptDebugPayload.mapping]:
```suggestion
CLASS net/minecraft/unmapped/C_djufnofa net/minecraft/network/packet/s2c/payload/WorldGenAttemptDebugPayload
```

**Review by supersaiyansubtlety — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/network/packet/s2c/payload/WorldgenAttemptDebugPayload.mapping (hunk 9 above)
Comment (supersaiyansubtlety) [reply mappings/net/minecraft/network/packet/s2c/payload/WorldgenAttemptDebugPayload.mapping]: The file name still has a lowercase g

Comment (eristhea) [reply mappings/net/minecraft/network/packet/s2c/payload/WorldgenAttemptDebugPayload.mapping]: fixed!

**Review by supersaiyansubtlety — APPROVED** [review]

**Pull request merged**